src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -576,6 +585,10 @@ protected void CheckModifiersAndType(BindingDiagnosticBag diagnostics)
             {
                 diagnostics.Add(ErrorCode.ERR_InstanceMemberInStaticClass, location, Name);
             }
+            else if (containingType.IsExtension && HasAssociatedField && !IsStatic)
+            {
+                diagnostics.Add(ErrorCode.ERR_StateInExtension, location, this);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_StateInExtension, location, this); [](http://example.com/codeflow?start=16&length=64)

Consider moving this check to the constructor (it has ```isFieldLike``` parameter). This will make it consistent with ERR_InterfacesCantContainFields reporting. This is where the similar check would be if it were possible to declare instance field-like events in interfaces.

Reply:
Perhaps it makes sense to do this after ```if (ContainingType.IsStatic && !IsStatic)```